dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WebSocketPublicStreamTestSuite.scala

```diff
@@ -0,0 +1,214 @@
+            asks = TreeMap.empty,
+            bids = TreeMap(0.00011403d -> 1.05d),
+            lastTrade = None,
+            timestamp = buffer1.last.timestamp
+          )
+        )
+
+        markup("Two orders")
+
+        placeAndAwaitAtDex(mkOrderDP(carol, wavesBtcPair, SELL, 1.waves, 0.00012))
+
+        val wsc2    = mkWebSocketOrderBookConnection(wavesBtcPair, dex1)
+        val buffer2 = receiveAtLeastN(wsc2, 1)
+        wsc2.close()
+
+        buffer2 should have size 1
+        squashOrderBooks(buffer2) should matchTo(
+          WsOrderBook(
+            asks = TreeMap(0.00012d    -> 1d),
+            bids = TreeMap(0.00011403d -> 1.05d),
+            lastTrade = None,
+            timestamp = buffer2.last.timestamp
+          )
+        )
+
+        markup("Two orders and trade")
+
+        placeAndAwaitAtDex(mkOrderDP(carol, wavesBtcPair, BUY, 0.5.waves, 0.00013), ApiOrderStatus.Filled)
+
+        val wsc3    = mkWebSocketOrderBookConnection(wavesBtcPair, dex1)
+        val buffer3 = receiveAtLeastN(wsc3, 1)
+        wsc3.close()
+
+        buffer3.size should (be >= 1 and be <= 2)
+        squashOrderBooks(buffer3) should matchTo(
+          WsOrderBook(
+            asks = TreeMap(0.00012d    -> 0.5d),
+            bids = TreeMap(0.00011403d -> 1.05d),
+            lastTrade = WsLastTrade(
+              price = 0.00012d,
+              amount = 0.5,
+              side = OrderType.BUY
+            ).some,
+            timestamp = buffer3.last.timestamp
+          )
+        )
+
+        markup("Four orders")
+
+        List(
+          mkOrderDP(carol, wavesBtcPair, SELL, 0.6.waves, 0.00013),
+          mkOrderDP(carol, wavesBtcPair, BUY, 0.7.waves, 0.000115)
+        ).foreach(placeAndAwaitAtDex(_))
+
+        val wsc4    = mkWebSocketOrderBookConnection(wavesBtcPair, dex1)
+        val buffer4 = receiveAtLeastN(wsc4, 1)
+        wsc4.close()
+
+        buffer4.size should (be >= 1 and be <= 2)
+        // TODO this test won't check ordering :(
+        squashOrderBooks(buffer4) should matchTo(
+          WsOrderBook(
+            asks = TreeMap(
+              0.00012d -> 0.5d,
+              0.00013d -> 0.6d,
+            ),
+            bids = TreeMap(
+              0.000115d   -> 0.7d,
+              0.00011403d -> 1.05d
+            ),
+            lastTrade = WsLastTrade(
+              price = 0.00012d,
+              amount = 0.5,
+              side = OrderType.BUY
+            ).some,
+            timestamp = buffer4.last.timestamp
+          )
+        )
+
+        dex1.api.cancelAll(carol)
+      }
+
+      "should send updates" in {
+        val firstOrder = mkOrderDP(carol, wavesBtcPair, BUY, 1.05.waves, 0.00011403)
+        placeAndAwaitAtDex(firstOrder)
+        dex1.api.cancelAll(carol)
+        dex1.api.waitForOrderStatus(firstOrder, ApiOrderStatus.Cancelled)
+
+        val wsc = mkWebSocketOrderBookConnection(wavesBtcPair, dex1)
+        receiveAtLeastN(wsc, 1)
+        wsc.clearMessagesBuffer()
```

Comment:
Looks like these placement and cancelling are not required by test context

Reply:
ok, removed